FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc

```diff
@@ -115,35 +97,16 @@ TrajectorySeedProducer::TrajectorySeedProducer(const edm::ParameterSet& conf):
         _seedingTree.insert(trackingLayerList);
         seedingLayers.push_back(std::move(trackingLayerList));
     }
-
-    originRadius = conf.getParameter<double>("originRadius");
-    originHalfLength = conf.getParameter<double>("originHalfLength");
-    ptMin = conf.getParameter<double>("ptMin");
-    nSigmaZ = conf.getParameter<double>("nSigmaZ");
-
-    //make sure that only one cut is configured
-    if (originHalfLength>=0 && nSigmaZ>=0)
-    {
-        throw cms::Exception("FastSimulation/Tracking/TrajectorySeedProducer: bad configuration","Either 'originHalfLength' or 'nSigmaZ' selection should be configured; not both. Deactivate one (or both) by setting it to <0.");
-    }
-
-    //make sure that performance cuts are not interfering with selection on reconstruction
-    if ((ptMin>=0 && simTrack_pTMin>=0) && (ptMin<simTrack_pTMin))
-    {
-        throw cms::Exception("FastSimulation/Tracking/TrajectorySeedProducer: bad configuration","Performance cut on SimTrack pT is tighter than cut on pT estimate from seed.");
-    }
-    if ((originHalfLength>=0 && simTrack_maxZ0>=0) && (originHalfLength>simTrack_maxZ0))
-    {
-        throw cms::Exception("FastSimulation/Tracking/TrajectorySeedProducer: bad configuration","Performance cut on SimTrack dz is tighter than cut on dz estimate from seed.");
-    }
-    if ((originRadius>=0 && simTrack_maxD0>=0) && (originRadius>simTrack_maxD0))
-    {
-        throw cms::Exception("FastSimulation/Tracking/TrajectorySeedProducer: bad configuration","Performance cut on SimTrack dxy is tighter than cut on dxy estimate from seed.");
+    if(conf.exists("RegionFactoryPSet")){
```

Comment:
don't do the 'if'
Assume RegionFactoryPSet is there.
If it's not there, cmsRun will crash, and that's how it's supposed to be.